zerver/lib/narrow.py

```diff
@@ -919,6 +927,30 @@ def get_channel_from_narrow_access_unchecked(
     return None
 
 
+def update_narrow_terms_containing_with_operator(
+    realm: Realm, narrow: OptionalNarrowListT
+) -> OptionalNarrowListT:
```

Comment:
Please add a nice docstring explaining this function; this feature is an important and somewhat strange thing we're doing, and so deserves some nice context-setting. It should probably mention the extra RTT considerations for why we're doing this in the server, for example.

Reply:
updated!